include/cinder/audio/Target.h

```diff
@@ -32,30 +32,42 @@ namespace cinder { namespace audio {
 
 typedef std::shared_ptr<class TargetFile>		TargetFileRef;
 
+namespace dsp {
+	class Converter;
+}
+
 //! Base class that is used to create and write to an audio destination. Currently only supports .wav encoding.
 class CI_API TargetFile {
   public:
-	static std::unique_ptr<TargetFile> create( const DataTargetRef &dataTarget, size_t sampleRate, size_t numChannels, SampleType sampleType = SampleType::INT_16, const std::string &extension = "" );
-	static std::unique_ptr<TargetFile> create( const fs::path &path, size_t sampleRate, size_t numChannels, SampleType sampleType = SampleType::INT_16, const std::string &extension = "" );
-	virtual ~TargetFile() {}
+	static std::unique_ptr<TargetFile> create( const DataTargetRef &dataTarget, size_t sampleRate, size_t numChannels, SampleType sampleType = SampleType::INT_16, size_t targetSampleRate = 0, const std::string &extension = "" );
+	static std::unique_ptr<TargetFile> create( const fs::path &path, size_t sampleRate, size_t numChannels, SampleType sampleType = SampleType::INT_16, size_t targetSampleRate = 0, const std::string &extension = "" );
+	virtual ~TargetFile();
 
 	void write( const Buffer *buffer );
 	void write( const Buffer *buffer, size_t numFrames );
 	void write( const Buffer *buffer, size_t numFrames, size_t frameOffset );
 
+	//! Returns the user facing sample rate (input)
 	size_t getSampleRate() const	{ return mSampleRate; }
+	//! Returns the true sample rate of the target file. \note Actual input samplerate may differ. \see getSampleRate()
+	size_t getSampleRateTarget() const	{ return mSampleRateTarget; }
```

Comment:
Do you think it's worth naming this `getSampleRateNative()` to be consistent with [`SourceFile`](https://github.com/cinder/Cinder/blob/649732c2868315ee5a606befdb2e9e43ff271732/include/cinder/audio/Source.h#L50)? I can't say I like one name over the other, though it is slightly ambiguous that the class is called `TargetFile` and the method has the noun 'target' in in.